### PR TITLE
Meson + Dynamic Caps

### PR DESCRIPTION
**Pull request by chriswiggins**
This PR converts building the plugin to using the meson build system. This is now the default for GStreamer 1.20 onwards now that they have the monorepo set up.

This also changes how caps are set on the src pad - instead of having 3 separate types of pads with SOMETIMES caps on each of them, this amalgamates all src types onto the one pad, and sets the caps when the data format is known

### CODE REVIEW

File: gst/milestonexprotect/meson.build [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Python 3 required modules
+python3_required_modules = ['gi']
+
+foreach p : python3_required_modules
+  # Source: https://docs.python.org/3/library/importlib.html#checking-if-a-module-can-be-imported
+  script = 'import importlib.util; import sys; exit(1) if importlib.util.find_spec(\''+ p +'\') is None else exit(0)'
+  if run_command(python3, '-c', script, check: false).returncode() != 0
+    error('Required Python3 module \'' + p + '\' not found')
+  endif
+endforeach
```
Comment (eli-schwartz) [inline gst/milestonexprotect/meson.build]: Since Meson 0.51 you can just use the `modules: [...]` kwarg to find_installation.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  '-Wmissing-include-dirs',
+  '-Waddress',
+  '-Waggregate-return',
+  '-Wno-multichar',
+  '-Wdeclaration-after-statement',
+  '-Wvla',
+  '-Wpointer-arith',
+]
+foreach extra_arg : warning_flags
+  if cc.has_argument (extra_arg)
```
Comment (eli-schwartz) [inline meson.build]: You can just use https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerget_supported_arguments here.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: gst/milestonexprotect/meson.build (hunk 1 above)
Comment (chriswiggins) [reply gst/milestonexprotect/meson.build]: Thank you! Will test and update

File: meson.build (hunk 2 above)
Comment (chriswiggins) [reply meson.build]: Thanks again!

**Pull request merged**